packages/react-components/src/components/chart/hooks/useViewportToMS.ts

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useState } from 'react';
+import { isDurationViewport, convertViewportToMs } from '../utils/getInfo';
+import { Viewport } from '@iot-app-kit/core';
+import { LIVE_MODE_REFRESH_RATE_MS } from '../eChartsConstants';
+
+export const useViewportToMS = (viewport?: Viewport) => {
+  const [inMS, setInMS] = useState(convertViewportToMs(viewport));
+
+  const viewportString = JSON.stringify(viewport);
+  useEffect(() => {
+    let interval: NodeJS.Timer;
+    if (viewport && isDurationViewport(viewport)) {
+      interval = setInterval(() => {
+        setInMS(convertViewportToMs(viewport));
+      }, LIVE_MODE_REFRESH_RATE_MS);
```

Comment:
refresh rate is configurable on the time series data hook, I think we should make those values are in sync since we will want to make that configurable through the viewport selector in the future.

Reply:
this is the chart refresh rate which is different from the api fetch frequency right ? 
the api frequency is 5s i think, and this will stay at 1sec is what i gathered from @tracy-french .. not sure, if we want to expose this as a configurable value...

Reply:
With the 5sec delay, for 10sec  viewport, the movement is not smooth, so needs to be 1000ms.